controllers/admin/AdminSuppliersController.php

```diff
@@ -565,4 +567,23 @@ protected function afterUpdate($object)
 
         return true;
     }
+
+    /**
+     * builds a message related to the display suppliers configuration status
+     *
+     * @return string
+     */
+    protected function supplierInfoMessage()
+    {
+        $message = sprintf(
+            'The display of your suppliers is %s on your store. Go to Shop Parameters > General to edit settings.',
+            Configuration::get('PS_DISPLAY_SUPPLIERS') ? 'enabled' : 'disabled'
+        );
+
+        return $this->trans(
```

Comment:
Is it okay if the wording does not follow the `$this->trans` here?

Reply:
Because the wording is dynamic, I build it first, inside the $message variable, then I use it inside $this->trans.
(is it what you meant ?)

Reply:
Alright, perfect, I don't know much about code so I wanted to ask just to be sure. ;-)

Reply:
Actually the `TranslationTool` won't be able to scan this message, because it can't `eval` the content of `$message` variable, si you'd better use an `if` condition and write two calls of `$this->trans` with hard coded strings

Reply:
I think it would be a good idea to integrate a link in the message, so that user can go directly on the appropriate page

Reply:
Also, with this the message will always be displayed, is there a way of closing this message?

Reply:
Ok, I had no idea the translation messages were scanned without executing the code...
I made the fix.

Reply:
About the link, yes I will add it ;)

Reply:
> Ok, I had no idea the translation messages were scanned without executing the code...
Indeed, of course I'm not talking about the `Translator` service, but `TranslatorTool` which we use to generate the XLF catalogs, it scan (almost) all the files of the projects (PHP, Twig, ...) looking code match `trans('my text')` or `'my text'|trans` and can then extract the translation keys. But of course it can not deduce the context that would generate a key if you integrated variables inside.

Reply:
About the link, you should use a key looking like this `To manage all your modules go to the [1]Installed module page[/1]` this prevents adding html code inside them

Reply:
Here's an example https://github.com/PrestaShop/PrestaShop/blob/e762a65d41ff00588bcdccb1ee196bc6c835d4cc/src/PrestaShopBundle/Resources/views/Admin/Product/ProductPage/product.html.twig#L197

Although I think you can inject the parameters directly to `trans` instead of using `|replace`

Reply:
done !

Reply:
yes it's better to use the trans parameter than replace, it's weird for the `[1]` we have a lot of them in our trads

Reply:
You didn't manage the link in this legacy controller, this will result in two translation keys (one with the link placeholders, the other one without them)

Reply:
This legacy controller is no longer used so I just removed what I did here.